src/bucket/BucketList.h

```diff
@@ -398,11 +398,15 @@ class BucketListDepth
     friend class testutil::BucketListDepthModifier;
 };
 
-class BucketList
+class BucketListBase
 {
+  protected:
     std::vector<BucketLevel> mLevels;
 
   public:
+    // Trivial pure virtual destructor to make this an abstract class
+    virtual ~BucketListBase() = 0;
```

Comment:
No this was intended. I want the base BucketList class to be pure virtual, but I define all the functions it declares. so I define a trivial pure virtual destructor.

Reply:
oh, I didn't realize this was the _only_ pure virtual function in this class. This feels a bit hacky to me. An abstract class implies there's an interface subclasses should be implementing, or they can re-use the default implementation in the base class (which means base class functions would be virtual). Looking at the interface, why is `addBatch` not pure virtual for example? Forcing subclasses to implement destructors seems a bit weird, since it's a default implementation anyway.